Automattic/jetpack
# [WIP] GIF Block: Replace source with Tumblr GIF Search

## Proposed changes:
* Updated the GIF block to use Tumblr's GIF search API, replacing the current Giphy-powered block.
* Added WP.com proxy endpoints for `/v2/gif/search/{query}` and `/v2/gif/popular` to retrieve Tumblr GIFs.
* Adds backwards compatibility so if a `giphyUrl` attribute is still detected, we honor those embeds.
* Updated tests.

### Other information:


## Jetpack product discussion
* pg1lMq-7O-p2
* pg1lMq-80-p2

## Does this pull request change what data or activity we track or use?
No.

## Testing instructions:

**Testing an existing GIPHY GIF**
* Before switching to this branch, create a new post or edit an existing post when running `trunk`
* Add the GIF block and insert it into the post.
* Enter a search term in the "Search" field.
* Select a GIF from GIPHY.
* Save or update the post.
* Switch Jetpack to use this branch.
* View the post on the frontend site.
* Check that the GIPHY GIF still displays correctly.

**Creating a new GIF**
* Run `bin/jetpack-downloader test jetpack tumblr-gif-block` to apply the endpoints to your sandbox.
* Set `JETPACK__SANDBOX_DOMAIN` to point at your sandbox.
* Create a new post or edit an existing post.
* Add the GIF block and insert it into the post.
* Enter a search term in the "Search" field.
* Select a GIF from Tumblr.
* Add a caption to the GIF (optional).
* Save or update the post.
* View the post on the frontend site.
* Check that the GIF displays correctly with the caption and attribution.

## Demo:

https://github.com/user-attachments/assets/88e0f694-8cb9-41fc-9281-7ed7563229de
